### DISCUSSION TITLE
How to get disk device IDs when nodes are in stage `Booting`?

### DISCUSSION THREAD

**Discussion by ringods** [siderolabs/talos#7960] [Q&A]
I'm in the process of netbooting a set of RPi's with Talos. The `install.disk` spec seems to be wrong in my setup. I have my node in the `Booting` stage. It has an IP address and connectivity says `OK`. Using `talosctl disks` errors on:

```sh
$ talosctl disks --talosconfig bootstrap-pxe-setup/assets/talos/talosconfig -e 172.16.40.142 -n 172.16.40.142
error getting disks: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 172.16.40.142:50000: connect: connection refused"
```

Can I retrieve disk info with my node in the `Booting` stage?

**Comment by smira** [comment]
You can either access Talos API in maintenance mode (when there's no machine config) using `--insecure`, or in "normal" mode when the machine config is already applied.

In the normal mode API is available after initial boot for controlplane node always, and for workers as soon as they are able to contact the controlplane.

So in your case it's either still booting up and you need to wait, or you have a worker which doesn't have a controlplane to talk to.

**Reply by ringods** [reply]
@smira I might be in a chicken-and-egg situation. See here the console view of the current state:

<img width="1349" alt="Screenshot 2023-11-12 at 12 10 30" src="https://github.com/siderolabs/talos/assets/77923/ea4500a9-b0d3-414d-bfcc-ec42517527d6">

The RPi that netbooted is fetching the controlplane config file, but due to the error on the wrong disk specification, I guess it doesn't end up in one of the 2 states you mentioned. In this state, I can wait indefinitely.

To find out which disk device IDs Talos detects, I probably have to do a manual install once and then read the `talosctl disks` info, right?

**Reply by smira** [reply]
In this state it tries to download machine config and fails to validate it. So you can either update on the server, or boot without talos.config argument first to make it enter maintenance mode, now you can figure out blockdevice name. If you have just one blockdevice, you can select it without knowing the name with device selectors.

Also the device name should be visible in the kernel logs if you scroll up.
